packages/server/src/service-worker.ts

```diff
@@ -135,13 +135,23 @@ function shouldDrop(request: Request, url: URL): boolean {
  * Forward request to main using the broadcast channel
  */
 async function broadcastOne(request: Request): Promise<Response> {
+  const message = await request.json();
   const promise = new Promise<Response>((resolve) => {
-    broadcast.onmessage = (event) => {
-      resolve(new Response(JSON.stringify(event.data)));
+    const messageHandler = (event: MessageEvent) => {
+      const data = event.data;
+      const browsingContextId = data.browsingContextId;
+      if (browsingContextId !== message.browsingContextId) {
+        // bail if the message is not for us
+        return;
+      }
+      const response = data.response;
+      resolve(new Response(JSON.stringify(response)));
+      broadcast.removeEventListener('message', messageHandler);
     };
+
+    broadcast.addEventListener('message', messageHandler);
```

Comment:
It looks like the previous logic was setting `broadcast.onmessage` for each request, which would override the function previously defined by an in-flight request.

`addEventListener` / `removeEventListener` should help handle messages from multiple closures, when several (concurrent) file system requests are in-flight.